src/common/download.cpp

```diff
@@ -40,27 +41,86 @@
 
 namespace tools
 {
-  static bool download_thread(const std::string &path, const std::string &url)
+  struct download_thread_control
   {
+    const std::string path;
+    const std::string uri;
+    std::function<void(const std::string&, const std::string&, bool)> result_cb;
+    std::function<bool(const std::string&, const std::string&, size_t, ssize_t)> progress_cb;
+    bool stop;
+    bool stopped;
+    bool success;
+    std::shared_ptr<boost::thread> thread;
```

Comment:
`std::shared_ptr<boost::thread>` -> `boost::thread`. Before C++11 moves there were certain contexts where `boost::thread` had to be heap allocated directly or indirectly because it cannot be copied, but that is no longer the case.